lib/mobility-core/src/Kernel/External/Payment/Interface/Juspay.hs

```diff
@@ -78,15 +80,44 @@ mandateNotification config req = do
   where
     mkNotificationRes Juspay.MandateNotificationRes {..} =
       MandateNotificationRes
+        { juspayProvidedId = id,
+          sourceInfo = castSourceInfo source_info,
+          notificationId = object_reference_id,
+          providerName = provider_name,
+          notificationType = notification_type,
+          description,
+          status,
+          dateCreated = posixSecondsToUTCTime $ fromIntegral (read (T.unpack date_created) :: Int),
```

Comment:
Can you use `readMaybe` or `readEither` and throw an error if parsing failed

Reply:
This will be done in seperate PR @aayushagarwal2001 will do it, for now I will merge this.